data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -136,7 +136,8 @@ public void start(Buffer<Record<Event>> buffer) {
                             break;
                         case PLAINTEXT:
                         default:
-                            if (sourceConfig.getSchemaConfig().getType() == SchemaRegistryType.AWS_GLUE) {
```

Comment:
minor: this is more readable IMO.
if (hasSchemaGlue()){
  getGlueDeserializer()
} else {
...